Sources/Paywalls/PaywallData.swift

```diff
@@ -190,7 +190,16 @@ extension PaywallData {
         public var defaultPackage: String?
 
         /// The images for this template.
-        public var images: Images
+        public var images: Images {
+            get {
+                return Self.merge(source: self._imagesHeic, fallback: self._legacyImages)
+            }
+
+            set {
+                self._imagesHeic = newValue
+                self._legacyImages = nil
```

Comment:
Also, when would this setter be called?

Reply:
This preserves the semantics of the `setter`.
Imagine a given configuration:
```swift
var config = Configuration(
  imagesHeic: Images(...),
  legacyImages: Images(...)
)
```

Doing this:
```swift
config.images.background = "new"
```

Is equivalent in Swift to this:
```swift
var newImages = config.images // This is getting the "derived" getter, merging both
newImages.background = "new"
config.images = newImages
```

This implementation is tested in a new test to verify its correctness too. Basically, not removing `_legacyImages` would mean you can't do this:
```swift
config.images.background = nil
```

Because it would potentially still find the fallback in `legacyImages`. This setter is mainly used for tests. Everywhere else we use immutable `PaywallData` so the setter isn't accessible anyway.

Reply:
I see! Thanks for taking the time to write that up